### PR TITLE
Fix Lua detection

### PR DESCRIPTION
**Pull request by berolinux**
- Detect Lua 5.3
- Fix detection of LuaJIT (before, it wouldn't set the correct
  variables and LuaJIT wouldn't be used even if there)

Signed-off-by: Bernhard Rosenkränzer <bero@lindev.ch>

### CODE REVIEW

File: 3party/osrm/osrm-backend/cmake/FindLua53.cmake [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+include(FindPackageHandleStandardArgs)
+# handle the QUIETLY and REQUIRED arguments and set LUA_FOUND to TRUE if
+# all listed variables are TRUE
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(Lua53
+                                  REQUIRED_VARS LUA_LIBRARIES LUA_INCLUDE_DIR
+                                  VERSION_VAR LUA_VERSION_STRING)
+
+mark_as_advanced(LUA_INCLUDE_DIR LUA_LIBRARIES LUA_LIBRARY LUA_MATH_LIBRARY)
+
```
Comment (Zverik) [inline 3party/osrm/osrm-backend/cmake/FindLua53.cmake]: Лишняя пустая строка пролезла.

**Comment by Zverik** [comment]
Could you please sign [the CLA](http://maps.me/cla/)?

**Comment by berolinux** [comment]
Sure, CLA is signed.

**Comment by Zverik** [comment]
Awesome. Please remove the empty line I commented on, and this could be merged then.

**Comment by berolinux** [comment]
Done. (Also removed the equivalent empty line from FindLua52.cmake, where my copy of it originated).

**Comment by Zverik** [comment]
Thank!

**Pull request merged**